src/lib/kafka/consumer.js

```diff
@@ -139,16 +131,68 @@ const registerNotificationHandler = async () => {
     }
     NotificationHandler.config.rdkafkaConf['client.id'] = NotificationHandler.topicName
     await createHandler(NotificationHandler.topicName, NotificationHandler.config)
+    await isConsumerConnected(NotificationHandler.topicName)
+
     return true
   } catch (e) {
     Logger.error(e)
     throw e
   }
 }
+/**
+ * @function getListOfTopics
+ *
+ * @description Get a list of topics that the consumer has subscribed to
+ *
+ * @returns {Array<string>} - list of topics
+ */
+const getListOfTopics = () => {
+  return Object.keys(listOfConsumers)
+}
+
+const getMetadataPromise = (consumer, topic) => {
+  return new Promise((resolve, reject) => {
+    const cb = (err, metadata) => {
+      if (err) {
+        return reject(new Error(`Error connecting to consumer`))
+      }
+
+      return resolve(metadata)
+    }
+
+    consumer.getMetadata({ topic, timeout: 3000 }, cb)
+  })
+}
+
+/**
+ * @function isConsumerConnected
+ *
+ * @param {string} topicName - the topic name of the consumer to check
+ *
+ * @description Use this to determine whether or not we are connected to the broker. Internally, it calls `getMetadata` to determine
+ * if the broker client is connected.
+ *
+ * @returns {true} - if connected
+ * @throws {Error} - if consumer can't be found or the consumer is not connected
+ */
+const isConsumerConnected = async topicName => {
```

Comment:
Same comment as the above naming convention. 

I feel like this is very much the same as a "double negative". A single negative is enough, where two is just verbose.

Let me know your thoughts here?

Reply:
Agreed. Thanks for pointing this out.